packages/flutter/lib/src/material/data_table.dart

```diff
@@ -0,0 +1,757 @@
+///
+/// One column configuration must be provided for each column to
+/// display in the table. The list of [DataColumn] objects is passed
+/// as the `columns` argument to the [new DataTable] constructor.
+class DataColumn {
+  /// Creates the configuration for a column of a [DataTable].
+  ///
+  /// The [label] argument must not be null.
+  const DataColumn({
+    this.label,
+    this.tooltip,
+    this.numeric: false,
+    this.onSort
+  });
+
+  /// The column heading.
+  ///
+  /// Typically, this will be either a [Text] widget, or a [Tooltip]
+  /// widget whose child is a [Text] widget. It could also include an
+  /// [Icon] (typically using size 18).
+  ///
+  /// The label should not include the sort indicator.
+  final Widget label;
+
+  /// The column heading's tooltip.
+  ///
+  /// This is a longer description of the column heading, for cases
+  /// where the heading might have been abbreviated to keep the column
+  /// width to a reasonable size.
+  final String tooltip;
+
+  /// Whether this column represents numeric data or not.
+  ///
+  /// Numeric data is presented by right-aligning the column.
+  final bool numeric;
+
+  /// Invoked when the user asks to sort the table using this column.
+  ///
+  /// If null, the column will not be considered sortable.
+  ///
+  /// See [DataTable.sortColumn] and [DataTable.sortAscending].
+  final DataColumnSortCallback onSort;
+
+  bool get _debugInteractive => onSort != null;
+}
+
+/// Row configuration and cell data for a [DataTable].
+///
+/// One row configuration must be provided for each row to
+/// display in the table. The list of [DataRow] objects is passed
+/// as the `rows` argument to the [new DataTable] constructor.
+///
+/// The data for this row of the table is provided in the [cells]
+/// property of the [DataRow] object.
+class DataRow {
+  /// Creates the configuration for a row of a [DataTable].
+  ///
+  /// The [cells] argument must not be null.
+  const DataRow({
+    this.key,
+    this.selected: false,
+    this.onSelectChanged,
+    this.cells
+  });
+
+  /// A [Key] that uniquely identifies this row. This is used to
+  /// ensure that if a row is added or removed, any stateful widgets
+  /// related to this row (e.g. an in-progress checkbox animation)
+  /// remain on the right row visually.
+  ///
+  /// If the table never changes once created, no key is necessary.
+  final LocalKey key;
+
+  /// Invoked when the user selects or unselects a selectable row.
+  ///
+  /// If this is not null, then the row is selectable. The current
+  /// selection state of the row is given by [selected].
+  ///
+  /// If any row is selectable, then the table's heading row will have
+  /// a checkbox that can be checked to select all selectable rows
+  /// (and which is checked if all the rows are selected), and each
+  /// subsequent row will have a checkbox to toggle just that row.
+  ///
+  /// Rows whose [onSelectChanged] callback is null are ignored for
```

Comment:
This is beyond my grammer skill level but maybe: A Row whose [onSelectChanged] callback is null is ignored for